components/DelegateCard.tsx

```diff
@@ -0,0 +1,166 @@
+        instructions,
+        realm.owner, // publicKey of program/programId
+        programVersion, // program version of realm
+        realm.pubkey, // realm public key
+        tokenMintToDelegateWith, // mint of governance token
+        wallet?.publicKey, // governingTokenOwner (walletId) publicKey of for tokenOwnerRecord of this wallet
+        wallet?.publicKey, // governanceAuthority: publicKey of connected wallet
+        new PublicKey(delegateKey) // public key of wallet who to delegated vote to
+      )
+      const recentBlockhash = await connection.getRecentBlockhash()
+      const transaction = new Transaction({
+        recentBlockhash: recentBlockhash.blockhash,
+      })
+      transaction.add(...instructions)
+      transaction.feePayer = wallet?.publicKey
+      await sendTransaction({ transaction, wallet, connection, signers })
+      setLoading(false)
+    } catch (error) {
+      console.log('error', error)
+    }
+  }
+
+  return (
+    <div className="bg-bkg-2 p-4 md:p-6 rounded-lg">
+      <h3 className="mb-4">Delegate tokens</h3>
+      {wallet && wallet.publicKey ? (
+        <>
+          <div className="text-sm text-th-fgd-1 flex flex-row items-center justify-between mt-4">
+            Allow any wallet to vote or create proposals with your deposited
+            tokens.
+          </div>
+          <div className="text-sm text-fgd-3">
+            This will not allow the delegated wallet to withdraw or send tokens.
+          </div>
+
+          <InputRow
+            label="Token Type"
+            icon={<CashIcon className="h-8 text-primary-light w-4 mr-2" />}
+          >
+            <div className="form-check">
+              <Checkbox
+                checked={tokenType === 'council'}
+                label={'Council Token'}
+                disabled={ownCouncilTokenRecord ? false : true}
+                onChange={() => setTokenType('council')}
+              />
+            </div>
+            <div className="form-check">
+              <Checkbox
+                checked={tokenType === 'community'}
+                label={'Community Token'}
+                disabled={ownTokenRecord ? false : true}
+                onChange={() => setTokenType('community')}
+              />
+            </div>
+          </InputRow>
+
+          <InputRow
+            label="Wallet"
+            icon={
+              <CreditCardIcon className="h-8 text-primary-light w-4 mr-2" />
+            }
+          >
+            <Input
+              className="w-full min-w-full"
+              type="text"
+              value={delegateKey}
+              onChange={(e) => setDelegateKey(e.target.value)}
+              placeholder="Public key"
+            />
+          </InputRow>
+
+          <Button
+            className="sm:w-full mt-4"
+            onClick={handleDelegate}
+            isLoading={isLoading}
+            disabled={
+              delegateKey.length !== 44 || (!ownTokenRecord && !ownTokenRecord)
```

Comment:
nit: ideally the validation would check if the key exists

Reply:
Switched to tryParseKey